app/src/main/res/xml/toggle_widget.xml

```diff
@@ -1,8 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <appwidget-provider xmlns:android="http://schemas.android.com/apk/res/android"
     android:initialLayout="@layout/widget_toggle"
-    android:minWidth="40dp"
-    android:minHeight="40dp"
+    android:minWidth="@dimen/widget_min_width"
+    android:minHeight="@dimen/widget_min_height"
+    android:minResizeWidth="@dimen/widget_min_resizable_width"
+    android:minResizeHeight="@dimen/widget_min_resizable_height"
+    android:previewImage="@drawable/widget_preview"
     android:resizeMode="horizontal|vertical"
     android:updatePeriodMillis="86400000"
```

Comment:
Is this update period how often the widget is going to sync with the new saved proxy?
If this is once a day and you change the proxy in the meantime from the app, 
will the widget show the right proxy?

Reply:
Yep, it's a required value when creating the widget to automatically refresh, set as once a day to save battery.
The proxy is updated via a broadcast sent every time you update the proxy from other sources (app, tile, etc)